File: src/main/java/com/chidozie/core/postProcessing/ContrastShader.java
package com.chidozie.core.postProcessing;

import com.chidozie.core.shaders.ShaderProgram;

public class ContrastShader extends ShaderProgram {

	private static final String VERTEX_FILE = "contrast/contrastVertexShader.vert";
	private static final String FRAGMENT_FILE = "contrast/contrastFragmentShader.frag";
	
	public ContrastShader() {
		super(VERTEX_FILE, FRAGMENT_FILE);
	}

	@Override
	protected void getAllUniformLocations() {	
	}

	@Override
	protected void bindAttributes() {
		super.bindAttribute(0, "position");
	}

}

File: src/main/kotlin/com/adv/core/postProcessing/ContrastShader.kt
package com.adv.core.postProcessing

import com.adv.core.shaders.ShaderProgram

class ContrastShader : ShaderProgram(VERTEX_FILE, FRAGMENT_FILE) {
    override fun getAllUniformLocations() {
    }

    override fun bindAttributes() {
        super.bindAttribute(0, "position")
    }

    companion object {
        private const val VERTEX_FILE = "contrast/contrastVertexShader.vert"
        private const val FRAGMENT_FILE = "contrast/contrastFragmentShader.frag"
    }
}
